scala/README.md

```diff
@@ -68,6 +67,8 @@ Please read the [Scala Style Guide] carefully. The main points to consider are:
 
 0. Public methods must always have explicit return types.
 
+    While it is not required to annotate non-public members, remember that explicitly declaring the return type allows the compiler to verify correctness.
```

Comment:
@matthandlersux I think the previous rule already covers that ("put clarity first and favor explicitness when an inferred type may not be obvious"). I'll change the examples to also show `def`s.